### ISSUE TITLE
make regex opt-in instead of default for string operations (`regex=False` instead of `Literal=False`)

### ISSUE THREAD

**Issue by JulianCologne** [pola-rs/polars#14801]
### Description

python and rust by default do not use regex in string operations.
You need another crate/package to achieve that (explicitly "opt-in")

I personally already had multiple bugs because this just does something unexpected

- python: `"hello.".replace(".", "!")` -> `"hello!"`
- rust: `"hello.".replace(".", "!")` -> `"hello!"`
- polars: `"pl.select(pl.lit("hello.").str.replace(".", "!"))"` -> `"!ello."` 😱 

Additionally polars replace using regex is also different to python re
- python (`import re`): `re.sub(".", "!", "hello.")` -> `"!!!!!!"`
- polars:  `"pl.select(pl.lit("hello.").str.replace(".", "!"))"` -> `"!ello."` 😱

**Comment by cmdlineluser** [comment]
Yep, I do recall ritchie previously inviting discussion for this: https://github.com/pola-rs/polars/issues/6120#issuecomment-1374860297

> Maybe we should make regex opt-in by adding a regex keyword argument? I am open to a discussion on this.

Just a personal opinion, but I do find having to put `regex=True` everywhere rather tiresome.

Would there be an argument for having separate regex functions?

```python
pl.col("foo").str.replace(old, new, regex=True)
pl.col("foo").str.regex_replace(old, new)
```
